octoprint_mrbeam/iobeam/iobeam_handler.py

```diff
@@ -512,22 +513,15 @@ def _work(self):
                             )
                             break
 
-                        # Processed buffered data as messages or skip and continue buffering the data
-                        if not data:
-                            my_errors = 1
-                            temp_buffer = b""
-                        else:
-                            # Split all JSON messages by new line character
-                            messages = data.split(self.MESSAGE_NEWLINE)
-
-                            if not data.endswith(self.MESSAGE_NEWLINE):
-                                # Record remaining part of data into temp buffer, to read messages longer than MESSAGE_LENGTH_MAX
-                                temp_buffer = messages.pop()
-                            else:
-                                temp_buffer = b""
+                        # Split all JSON messages by new line character
+                        messages = data.split(self.MESSAGE_NEWLINE)
```

Comment:
what exactly is the difference here?

Reply:
You already made a check if data existed or not.